src/tests.rs

```diff
@@ -1808,6 +1811,14 @@ fn mismatched_incompatible_bookmark_kinds() {
 fn invalid_guids() {
     before_each();
 
+    struct AllowInvalidGuids;
```

Comment:
Even though it's "just" test code I think a comment here might be useful to help people get their head around the intent behind `generate_new_guid()`